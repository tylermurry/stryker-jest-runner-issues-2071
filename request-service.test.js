jest.doMock('node-fetch', () => require('fetch-mock').sandbox());

const fetch = require('node-fetch');
const { request } = require('./request-service');

describe('Request Service Tests', () => {
  beforeEach(() => fetch.reset());

  it('should return the correct JSON response for a GET request with no options', async () => {
    fetch.get('http://someurl.com/', { some: 'get data' });

    expect(await request('http://someurl.com/')).toMatchSnapshot();
  });

  it('should return an empty body for a no-content response', async () => {
    fetch.get('http://someurl.com/', 204);

    expect(await request('http://someurl.com/')).toMatchSnapshot();
  });

  it('should return the correct JSON response for a POST request with options', async () => {
    fetch.post('http://someurl.com/', { some: 'post data' });

    expect(await request('http://someurl.com/', { method: 'POST', body: 'something' })).toMatchSnapshot();
  });

  it('should return the correct error payload if the request fails', async () => {
    fetch.get('http://someurl.com/', { status: 500, body: 'some error message' });

    await expect(request('http://someurl.com/')).rejects.toThrowErrorMatchingSnapshot();
  });

  it('should return the correct error payload if the request fails and masks the authorization header properly', async () => {
    fetch.get('http://someurl.com/', { status: 500, body: 'some error message' });

    await expect(request(
      'http://someurl.com/',
      { headers: { Authorization: "Bearer sometoken" }}
    )).rejects.toThrowErrorMatchingSnapshot();
  });
});
