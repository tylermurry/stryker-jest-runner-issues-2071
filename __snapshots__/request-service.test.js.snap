// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Service Tests should return an empty body for a no-content response 1`] = `
Object {
  "body": null,
  "rawResponse": Response {
    "size": 0,
    "timeout": 0,
    Symbol(Body internals): Object {
      "body": null,
      "disturbed": true,
      "error": null,
    },
    Symbol(Response internals): Object {
      "counter": undefined,
      "headers": Headers {
        Symbol(map): Object {},
      },
      "status": 204,
      "statusText": "No Content",
      "url": "http://someurl.com/",
    },
  },
}
`;

exports[`Request Service Tests should return the correct JSON response for a GET request with no options 1`] = `
Object {
  "body": Object {
    "some": "get data",
  },
  "rawResponse": Response {
    "size": 0,
    "timeout": 0,
    Symbol(Body internals): Object {
      "body": Object {
        "data": Array [
          123,
          34,
          115,
          111,
          109,
          101,
          34,
          58,
          34,
          103,
          101,
          116,
          32,
          100,
          97,
          116,
          97,
          34,
          125,
        ],
        "type": "Buffer",
      },
      "disturbed": true,
      "error": null,
    },
    Symbol(Response internals): Object {
      "counter": undefined,
      "headers": Headers {
        Symbol(map): Object {
          "Content-Length": Array [
            "19",
          ],
          "Content-Type": Array [
            "application/json",
          ],
        },
      },
      "status": 200,
      "statusText": "OK",
      "url": "http://someurl.com/",
    },
  },
}
`;

exports[`Request Service Tests should return the correct JSON response for a POST request with options 1`] = `
Object {
  "body": Object {
    "some": "post data",
  },
  "rawResponse": Response {
    "size": 0,
    "timeout": 0,
    Symbol(Body internals): Object {
      "body": Object {
        "data": Array [
          123,
          34,
          115,
          111,
          109,
          101,
          34,
          58,
          34,
          112,
          111,
          115,
          116,
          32,
          100,
          97,
          116,
          97,
          34,
          125,
        ],
        "type": "Buffer",
      },
      "disturbed": true,
      "error": null,
    },
    Symbol(Response internals): Object {
      "counter": undefined,
      "headers": Headers {
        Symbol(map): Object {
          "Content-Length": Array [
            "20",
          ],
          "Content-Type": Array [
            "application/json",
          ],
        },
      },
      "status": 200,
      "statusText": "OK",
      "url": "http://someurl.com/",
    },
  },
}
`;

exports[`Request Service Tests should return the correct error payload if the request fails 1`] = `
"{
  \\"request\\": {
    \\"url\\": \\"http://someurl.com/\\"
  },
  \\"response\\": {
    \\"status\\": 500,
    \\"body\\": \\"some error message\\"
  }
}"
`;

exports[`Request Service Tests should return the correct error payload if the request fails and masks the authorization header properly 1`] = `
"{
  \\"request\\": {
    \\"url\\": \\"http://someurl.com/\\",
    \\"options\\": {
      \\"headers\\": {
        \\"Authorization\\": \\"***\\"
      }
    }
  },
  \\"response\\": {
    \\"status\\": 500,
    \\"body\\": \\"some error message\\"
  }
}"
`;
